clippy_lints/src/contains_or_insert.rs

```diff
@@ -113,7 +118,7 @@ fn find_insert_calls<'tcx>(
     expr: &'tcx Expr<'_>,
 ) -> Option<OpExpr<'tcx>> {
     for_each_expr(cx, expr, |e| {
-        if let Some(insert_expr) = try_parse_op_call(cx, e, sym!(insert))
+        if let Some((insert_expr, _)) = try_parse_op_call(cx, e, sym!(insert))
             && SpanlessEq::new(cx).eq_expr(contains_expr.receiver, insert_expr.receiver)
```

Comment:
Here we should use `eq_expr_value` to avoid linting on e.g. `my_vec.pop().contains(key)`.

Reply:
@llogiq i must still be missing something. I did not change any comparison longic, but instead simply ignore one extra param.  If you think this lint should be modified to use `eq_expr_value` somewhere where my PR is not touching, lets perhaps do it in a separate PR?  Especially if you think there is a false positive in it that we should add to a test.

In the mean time, I reverted the lint rename - the cut off day has passed i think, so it would be better to keep the lint as is, and perhaps introduce another lint that focuses on `map` style interfaces (e.g. `if !hashmap.contains(key) { hashmap.add(key, value) }`)